### PR TITLE
Improve diagnostics for the case where some or all of the prelude is missing.

### PR DESCRIPTION
**Pull request by zygoloid**

### CODE REVIEW

**Comment by zygoloid** [comment]
Factored out of #4209.

File: toolchain/check/testdata/packages/no_prelude/missing_prelude.carbon [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+// --- prelude_as_namespace.carbon
+
+// TODO: Decide whether we want to accept this.
+namespace Core;
+
+fn Core.Int32() -> {} { return {}; }
+
+var n: {} = i32;
+
```
Comment (jonmeow) [inline toolchain/check/testdata/packages/no_prelude/missing_prelude.carbon:73]:
```suggestion

// --- fail_prelude_as_class.carbon

// TODO: Decide whether we want to accept this.
class Core {
  fn Int32() -> {} { return {}; }
}

var n: {} = i32;

```

File: toolchain/check/testdata/packages/no_prelude/missing_prelude.carbon [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+// --- prelude_use_in_prelude.carbon
+
+package Core library "prelude_use_in_prelude";
+
+// Core is not an imported package here.
+
+fn Int32() -> () { return (); }
+
+var n: () = i32;
```
Comment (jonmeow) [inline toolchain/check/testdata/packages/no_prelude/missing_prelude.carbon:61]: Other tests use `{}`, so suggesting for consistency.

```suggestion
fn Int32() -> {} { return {}; }

var n: {} = i32;
```

**Review by jonmeow — APPROVED** [review]
Thanks!

File: toolchain/check/testdata/packages/no_prelude/missing_prelude.carbon (hunk 2 above)
Comment (zygoloid) [reply toolchain/check/testdata/packages/no_prelude/missing_prelude.carbon:61]: I was using a different type here to ensure we were picking the definition from this file, not from the other library.

**Pull request merged**